src/Apps/Order/Components/PaymentPicker.tsx

```diff
@@ -197,6 +197,16 @@ export class PaymentPicker extends React.Component<
 
     const userHasExistingCards = creditCards.edges.length > 0
 
+    const orderCard = this.props.order.creditCard
+
+    let defaultCard
+
+    if (creditCardSelection.type === "new") {
+      defaultCard = "new"
+    } else {
+      defaultCard = orderCard ? orderCard.id : creditCardSelection.id
+    }
+
```

Comment:
This looks good, but I think it makes more sense to put this logic in the `getInitialCreditCardSelection` method near the top of this file.